back/models/game.js

```diff
@@ -30,6 +30,25 @@ class Game {
     this.activePlayerId = this.players[nextPlayerIndex].id;
   }
 
+  validatePlayerTurn(playerId, card = null) {
+    const player = this.getPlayerById(playerId);
+    const cardStackCount = this.cardStack.length;
+    const lastCard = this.cardStack[cardStackCount - 1];
+    if (player.id !== this.activePlayerId) {
+      return false;
+    }
+    if (card) {
+      if (!player.cards.find(cardOnHand => isEqual(card, cardOnHand))) {
+        return false;
+      }
+      if (cardStackCount > 0 && 
+          (card.figure !== lastCard.figure && card.color !== lastCard.color )) {
```

Comment:
Logical `or` should be used here

Reply:
use `isEqual`

Reply:
Changed to ``` or ``` . Cannot use as it is not appropriate for the task.